### ISSUE TITLE
Add Sentry to pkgdb

### ISSUE THREAD

**Issue by zmitchell** [flox/flox#1056]
Add Sentry integration to `pkgdb` using the `sentry-native` library. This includes passing the following pieces of information from the Rust side:
- The Sentry DSN
- The Sentry environment
- The Sentry trace ID
- The current span ID (the "parent ID" of the `pkgdb` invocation)

This allows metrics from `flox` and `pkgdb` to be correlated.

**Acceptance criteria:**
- [ ] `pkgdb` initializes Sentry at startup
- [ ] `flox` sets `_FLOX_SENTRY_DSN`, `_FLOX_SENTRY_ENV`, `_FLOX_SENTRY_TRACE_ID`, and `_FLOX_SENTRY_PARENT_ID` environment variables for all `pkgdb` invocations
- [ ] `pkgdb` initializes Sentry with the values provided by `flox` if they are present

**Comment by billlevine** [comment]
This is now functional and can submit crashes and events, however in doing so, some complications have arisen and we may want to re-evaluate the scope of this work.

- Sentry 0.6.7 native SDK is in Nix for Linux but I'm getting an error on Darwin that it's not available on that system.  I'm assuming we could add it, but the fact that it's not leads me to be suspicious that *there be dragons*.
- Debug Symbols - The stack traces submitted have limited utility without debug symbols.  These can be uploaded to Sentry during build or release cycles, or we could setup a symbol server to house them and direct Sentry to resolve symbols via that method.  I think the later may be cleaner for a long term solution, but either would work.  (Ref)[https://docs.sentry.io/platforms/native/data-management/debug-files/]
- Versioning - Simpler but still something to resolve... The main flox version is on the Rust side and I don't see version info in `pkgdb`.  We should likely add that to mirror the main flox version since they are built in tandem.  Or pass it from `flox` via env var or something.
- Deployment on Mac could be more complicated... see _Deployment Notes_ below.

Deployment Notes:
```
Bundling crashpad_handler
When using the Crashpad backend, which is the default on Windows and macOS, the crashpad_handler binary has to be shipped alongside the application binary. See the [Crashpad documentation](https://docs.sentry.io/platforms/native/configuration/backends/crashpad/) for more information.
```

In summary:

As of right now, we could have crash reports with stack traces of limited utility only on Linux (1pt).  Just fix the version issue and it's something that could add some value.  Adding symbol file management would add utility to the diagnostics (3pts?).  Fixing the Mac build issue AND looking at the deployment issue would get us visibility on Mac (5pts across several individuals?).

IMO - getting crash reports on linux is better than nothing and we're basically there.
